core/src/types/vote_extensions/bridge_pool_roots.rs

```diff
@@ -0,0 +1,99 @@
+//! Vote extension types for adding a signature
+//! of the bridge pool merkle root to be added
+//! to storage. This will be used to generate
+//! bridge pool inclusion proofs for Ethereum.
+use std::collections::HashSet;
+use std::ops::{Deref, DerefMut};
+
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+
+use crate::proto::Signed;
+use crate::types::address::Address;
+use crate::types::key::common;
+use crate::types::key::common::Signature;
+use crate::types::storage::BlockHeight;
+
+/// A vote extension containing a validator's signature
+/// of the current root and nonce of the
+/// Ethereum bridge pool.
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    PartialOrd,
+    Eq,
+    Hash,
+    BorshSerialize,
+    BorshDeserialize,
+    BorshSchema,
+)]
+pub struct BridgePoolRootVext {
+    /// The validator sending the vote extension
+    /// TODO: the validator's address is temporarily being included
+    /// until we're able to map a Tendermint address to a validator
+    /// address (see <https://github.com/anoma/namada/issues/200>)
+    pub validator_addr: Address,
+    /// The block height at which the vote extensions was
+    /// sent.
+    ///
+    /// This can be used as replay protection as well
+    /// as allowing validators to  query the epoch with
+    /// the appropriate validator set to verify signatures
+    pub block_height: BlockHeight,
+    /// The actual signature being submitted.
+    /// This is a signature over KeccakHash(root || nonce).
```

Comment:
the docstring mentions `KeccakHash(root || nonce)`, which is not what is being signed. we're only signing the inner contents, i.e. `root || nonce`

Reply:
technically `eth_header || root || nonce`, or `eth_header || KeccakHash(root || nonce)` in case we hash the data to be signed